src/backend/cuda/kernel/scan_first.hpp

```diff
@@ -145,13 +143,12 @@ namespace kernel
         Binary<To, op> binop;
         To accum = tptr[blockIdx_x - 1];
 
+        int offset = !inclusive_scan;
         for (int k = 0, id = xid;
-             k < lim && id < out.dims[0];
+             k < lim && id + offset < out.dims[0];
```

Comment:
Increment id during initialization instead of incrementing it every time its used.